packages/react/src/hooks/notifyStateManagement.ts

```diff
@@ -3,18 +3,24 @@ import { useSubscriptionState } from "@web3inbox/core";
 import { useCallback, useEffect, useState } from "react";
 import { useWeb3InboxClient } from "./web3inboxClient";
 
-export const useMessages = (account?: string) => {
+ /**
+  * Hook to watch messages of a subscription, and delete them
+  *
+  * @param {string} [account] - Account to get subscriptions messages from , defaulted to current account
+  * @param {string} [domain] - Domain to get subscription messages from, defaulted to one set in init.
+  */
+export const useMessages = (account?: string, domain?: string)  => {
   const client = useWeb3InboxClient();
   const { messages: messagesTrigger } = useSubscriptionState();
   const [messages, setMessages] = useState<
     NotifyClientTypes.NotifyMessageRecord[]
-  >(client?.getMessageHistory(account) ?? []);
+  >(client?.getMessageHistory(account, domain) ?? []);
 
   useEffect(() => {
     if (!client) return;
 
     setMessages(client.getMessageHistory(account));
-  }, [client, messagesTrigger, account]);
+  }, [client, messagesTrigger, account, domain]);
```

Comment:
Based on some recent learnings, I think it's better to derive state instead.
It improves readability and avoids an additional render cycle.

```ts
const messages = client?.getMessageHistory(account, domain) ?? []
```

Reply:
Oh right, then we definitely need `useEffect` to sync it.
I wonder if we could use this [useSyncExternalStore](https://react.dev/reference/react/useSyncExternalStore) ? It feels like we have a use case for it. Wdyt ?

Reply:
this might come in handy when we refactor to have 3 separate packages I think :thinking:  didn't know about this. But definitely not something I want to risk implementing now considering the time contraints

Reply:
For sure, I was just throwing the idea. Great job on the PR 💪